### PR TITLE
Fix password toggle in dynamic subform field

### PR DESCRIPTION
**Pull request by MatMyfta**
Pull Request for Issue #43240 .

### Summary of Changes

Implemented a new DOM change event that activates the password toggle (show/hide) functionality, allowing it to operate effectively for both dynamic and static form fields.

### Testing Instructions

described in the issue.

### Actual result BEFORE applying this Pull Request

described in the issue.

### Expected result AFTER applying this Pull Request

The password toggle function now operates as expected, enabling correct visibility control for password fields in dynamic subforms.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: build/media_source/system/js/fields/passwordview.es6.js [hunk 1]
```diff
@@ -5,78 +5,97 @@
...
+  }
+  new MutationObserver((mutationsList) => {
+    document.dispatchEvent(new CustomEvent('domChanged', { detail: mutationsList }));
+  }).observe(document, {
+    attributes: true,
+    childList: true,
+    subtree: true,
+  });
+
+  document.addEventListener('domChanged', () => {
```
Comment (dgrammatiko) [inline build/media_source/system/js/fields/passwordview.es6.js]: You don't have to setup a mutation observer here, as the repeatable form field has it's own event: https://github.com/joomla/joomla-cms/blob/2d4547ea621e84408cbae8e20db9b05e192ca34f/build/media_source/system/js/fields/color-field-adv-init.es5.js#L33-L36

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

File: build/media_source/system/js/fields/passwordview.es6.js (hunk 1 above)
Comment (MatMyfta) [reply build/media_source/system/js/fields/passwordview.es6.js]: Thank you for the feedback.
I have tested the code using this event and works. 

I'm  updating now.

File: build/media_source/system/js/fields/passwordview.es6.js [hunk 2]
```diff
@@ -84,15 +84,8 @@
...
-  new MutationObserver((mutationsList) => {
-    document.dispatchEvent(new CustomEvent('domChanged', { detail: mutationsList }));
-  }).observe(document, {
-    attributes: true,
-    childList: true,
-    subtree: true,
-  });
 
-  document.addEventListener('domChanged', () => {
+  document.addEventListener('joomla:updated', () => {
```
Comment (dgrammatiko) [inline build/media_source/system/js/fields/passwordview.es6.js]: Assign the function directly 
```
document.addEventListener('joomla:updated', togglePassword);
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 8aacd6f2e797f2741676db243f8140980030f259<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44362">issues.joomla.org/tracker/joomla-cms/44362</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 8aacd6f2e797f2741676db243f8140980030f259<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44362">issues.joomla.org/tracker/joomla-cms/44362</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44362">issues.joomla.org/tracker/joomla-cms/44362</a>.</sub>

**Comment by Hackwar** [comment]
Thank you for your contribution!

**Comment by brianteeman** [comment]
Just in case someone else comes across this - you may need to re-create any mod_logged overrides - i did

**Pull request merged**